packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/palette/DiagramElementPalettePortal.tsx

```diff
@@ -12,12 +12,12 @@
  *******************************************************************************/
 import { ReactNode } from 'react';
 import { createPortal } from 'react-dom';
-import { ReactFlowState, useStore } from 'reactflow';
-
-const selector = (state: ReactFlowState) => state.domNode?.querySelector('.react-flow__renderer');
+import { useStoreApi } from 'reactflow';
 
 export const DiagramElementPalettePortal = ({ children }: { children: ReactNode }) => {
-  const wrapperRef = useStore(selector);
+  const store = useStoreApi();
+  const { domNode } = store.getState();
+  const wrapperRef = domNode?.querySelector('.react-flow__renderer');
```

Comment:
What is the point of this change?

Reply:
When using useStore() we subscribe to the state and the component will re render each time the state change. 
With useStoreApi() we get the value when needed.
With this change we avoid unnecessary rendering.